inc/shortcodes/class-iframe.php

```diff
@@ -49,6 +49,29 @@ public static function get_whitelisted_iframe_domains() {
 		return apply_filters( 'shortcake_bakery_whitelisted_iframe_domains', array() );
 	}
 
+	/**
+	 * Transform any <iframe> embeds within content to our iframe shortcode
+	 *
+	 * @param string $content
+	 * @return string
+	 */
+	public static function reversal( $content ) {
+
+		if ( $iframes = self::parse_iframes( $content ) ) {
+			$whitelisted_iframe_domains = static::get_whitelisted_iframe_domains();
+			$replacements = array();
+			foreach ( $iframes as $iframe ) {
+				if ( ! in_array( parse_url( $iframe->src_force_protocol, PHP_URL_HOST ), $whitelisted_iframe_domains ) ) {
+					continue;
+				}
+				$replacements[ $iframe->original ] = '[' . self::get_shortcode_tag() . ' src="' . esc_url_raw( $iframe->attrs['src'] ) . '"]';
```

Comment:
Hmm. Most of the attributes that we'll see will likely be fairly common styling options (`frameborder`, `border`, `scrolling`, `marginwidth`, `marginheight`) that we could store as a single attribute like 'iframe-style'.

The other attributes could probably be dealt with by a whitelist for now: just expose `id`, `class`, and maybe any others that are obvious... 

Persisting additional attributes might have to wait until https://github.com/fusioneng/Shortcake/issues/425 is fixed. Once that issue can be addressed, there wouldn't be any problem with taking any attributes that pass a `wp_kses_attr_check` and putting them in the shortcode, knowing that they won't be editable through the UI.